erpnext/hr/doctype/additional_salary/additional_salary.py

```diff
@@ -45,7 +58,11 @@ def get_additional_salary_component(employee, start_date, end_date):
 		from `tabAdditional Salary`
 		where employee=%(employee)s
 			and docstatus = 1
-			and payroll_date between %(from_date)s and %(to_date)s
+			and (payroll_date between %(from_date)s and %(to_date)s)
+			or (
+				(from_date between %(from_date)s and %(to_date)s)
+				or(to_date between %(from_date)s and %(to_date)s)
```

Comment:
it will give error if we make to_date optional

Reply:
made both mandatory if ``` is_recurring = 1 ```